webapp/src/projects.tsx

```diff
@@ -328,44 +338,29 @@ export class ProjectsCarousel extends data.Component<ProjectsCarouselProps, Proj
         return headers;
     }
 
-    newProject() {
-        pxt.tickEvent("projects.new");
-        this.props.hide();
-        this.props.parent.newProject();
-    }
-
     renderCore() {
         const {name, path} = this.props;
         const theme = pxt.appTarget.appTheme;
 
-        // Fetch the gallery
-        this.hasFetchErrors = false;
-        let cards: pxt.CodeCard[];
-        let headers: pxt.workspace.Header[];
-        if (path) {
-            cards = this.fetchGallery(path);
-        } else {
-            headers = this.fetchLocalData();
-            headers.unshift({
-                id: 'new',
-                name: lf("New Project")
-            } as any)
-        }
-
         const onClick = (scr: any) => {
             this.props.onClick(scr);
         }
 
-        return <div className="ui dimmable">
-            {this.hasFetchErrors ?
-                <div className="ui carouselouter">
+        let carouselJSX: JSX.Element;
```

Comment:
Why the variable? Why not just have separate return statements?

Reply:
Very different code paths whether its the header or parts. If split up, it would be three different chained if elses

Reply:
Ah I see what you mean. Yes. Fixing